glommio/benches/competing_io.rs

```diff
@@ -0,0 +1,146 @@
+use futures::future::join;
+use futures_lite::{stream, AsyncWriteExt, StreamExt};
+use glommio::{
+    enclose,
+    io::{ImmutableFile, ImmutableFileBuilder},
+    Latency,
+    LocalExecutorBuilder,
+    Placement,
+    Shares,
+};
+use rand::Rng;
+use std::{
+    cell::RefCell,
+    path::PathBuf,
+    rc::Rc,
+    time::{Duration, Instant},
+};
+
+fn main() {
+    let handle =
+        LocalExecutorBuilder::new(Placement::Fixed(0))
+            .spawn(|| async move {
+                let filename = match std::env::var("GLOMMIO_TEST_POLLIO_ROOTDIR") {
+                    Ok(path) => {
+                        let mut path = PathBuf::from(path);
+                        path.push("benchfile");
+                        path
+                    }
+                    Err(_) => panic!("please set 'GLOMMIO_TEST_POLLIO_ROOTDIR'"),
+                };
+
+                {
+                    let _ = std::fs::remove_file(&filename);
+                    let mut sink = ImmutableFileBuilder::new(&filename)
+                        .with_buffer_size(512 << 10)
+                        .with_pre_allocation(Some(64 << 20))
+                        .build_sink()
+                        .await
+                        .unwrap();
+
+                    let contents = vec![1; 512 << 10];
+                    for _ in 0..(64 << 20) / contents.len() {
+                        sink.write_all(&contents).await.unwrap();
+                    }
+                    let file = sink.seal().await.unwrap();
+
+                    let lat_tq = glommio::executor().create_task_queue(
+                        Shares::Static(1000),
+                        Latency::Matters(Duration::from_millis(1)),
+                        "lat",
+                    );
+                    let throughput_tq = glommio::executor().create_task_queue(
+                        Shares::Static(1000),
+                        Latency::NotImportant,
+                        "throughput",
+                    );
+                    let gate = Rc::new(RefCell::new(true));
```

Comment:
given the name, could you use https://docs.rs/glommio/0.6.0/glommio/sync/struct.Gate.html ?
If not, at least use a `Cell` instead of a `RefCell`